batch/batch/aioclient.py

```diff
@@ -152,36 +212,30 @@ def __init__(self, client, id, attributes):
         if pvc_size:
             doc['pvc_size'] = pvc_size
 
-        j = await self._client._post('/jobs/create', json=doc)
+        self._job_docs.append(doc)
 
-        return Job(self,
-                   j['job_id'],
-                   attributes=j.get('attributes'),
-                   parent_ids=j.get('parent_ids', []))
+        j = Job(self._batch,
```

Comment:
Similarly, this really isn't about creating a Job...it's about populating a doc. Without the ids, the Job doesn't really exist. 

I realize what you're going after, but it feels slightly wrong to have both a dirty Batch and Job (not synced with the db), and a separate store for the document shape (doc/self._doc).  

I would simple have no self._batch Jobs, until submit is called, which returns the batch (with id) and all Jobs.

Submit also seems slightly dangerous because you're mutating self._batch (adding the id), which updates the copy held in Job (which holds a pointer to the batch object), which means that submit has a fairly important side effect. It would be imo nicer to just have no side effects, and construct the objects there.